src/codemagic/cli/cli_app.py

```diff
@@ -104,16 +104,23 @@ def _parent_class_kwargs(cls, cli_args: argparse.Namespace) -> Dict[str, Any]:
         return {key: value for key, value in cli_args.__dict__.items() if key not in class_arg_keys}
 
     @classmethod
-    def _handle_generic_exception(cls, action_name: str) -> int:
-        logger = log.get_logger(cls)
+    def _handle_generic_exception(cls, args: argparse.Namespace) -> int:
+        subaction_name = getattr(args, "action_subcommand", None)
+        if subaction_name:
+            executed_command = f"{cls.get_executable_name()} {args.action} {subaction_name}"
+        else:
+            executed_command = f"{cls.get_executable_name()} {args.action}"
+
+        verbose_log_suggestion = 'To see more details about the error, add "--verbose" command line option.'
         message = (
-            f"Executing {cls.__name__} action {action_name} failed unexpectedly. "
+            f'Executing "{executed_command}" failed unexpectedly. '
             f'Detailed logs are available at "{log.get_log_path()}". '
-            f"To see more details about the error, add `--verbose` command line option."
+            f'{"" if args.verbose else verbose_log_suggestion}'
         )
-        logger.warning(Colors.RED(message))
-        file_logger = log.get_file_logger(cls)
-        file_logger.exception("Exception traceback:")
+
+        log.get_logger(cls).warning(Colors.RED(message))
+        logger = log.get_logger(cls, log_to_file=True, log_to_stream=args.verbose)
+        logger.exception("Exception traceback:")
         auditing.save_exception_audit()
```

Comment:
Worth perhaps move
```py
logger.exception("Exception traceback:")
```

inside `auditing.save_exception_audit()`

And maybe even
```py
logger = log.get_logger(cls, log_to_file=True, log_to_stream=args.verbose)
```

Reply:
I don't agree with this. Currently exception logging and the preceding message are somewhat bound together:
- if verbose output is enabled, then part of the previous message is omitted and stdout stream logging is enabled,
- or if verbose output is disabled, then stdout logging is turned off and there is a statement that more details can be seen if `--verbose` is used.
In case `logger.exception` statement is moved to somewhere else, then this logic becomes disconnected and harder to maintain.